File: 2022/javascript/day_1/day_1_2.js
const {readFileSync} = require("fs")

const input = readFileSync("./input.txt", "utf-8").replace(/\r/g, "")
const elves = input.split("\n\n")

const elves_sum = elves.map(e => {
    const nums = e.split("\n").map(n => Number(n)).reduce((a, b) => a + b)
    return nums
}).sort((a, b) => b - a)

console.log(elves_sum[0] + elves_sum[1] + elves_sum[2])

File: 2022/python/day_01/day_1_2.py
import functools
inp = open("input.txt", "r").read().replace(r"\r\n", "").split("\n\n")

values = list(map((lambda elf: list(map((lambda food: int(food)), elf.split("\n")))), inp))

total_calories = list(map(lambda elf: functools.reduce(lambda a, b: a+b, elf), values))
total_calories.sort(reverse=True)

print(f'The sum of the fullest backpacks is {functools.reduce(lambda a, b: a+b, total_calories[0:3])}')